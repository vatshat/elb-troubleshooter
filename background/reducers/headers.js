export default (state=[], action) => {
    switch (action.type) {
    case 'ADD_HEADER': return [...state, { 
        visible: false,
        headers: action.payload,         
    }];

    default: 
        return state;
    }
};