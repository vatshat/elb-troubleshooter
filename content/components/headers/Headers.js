import React, { Component } from 'react';
import { connect } from 'react-redux';

class App extends Component {
    constructor(props) {
        super(props);
    }

    componentDidMount() {
        document.addEventListener('click', () => {
            this.props.dispatch({
                type: 'ADD_COUNT'
            });
        });
    }
    
    openApp() {
        chrome.tabs.create({
            url: chrome.runtime.getURL('index.html')
        });
    }

    render() {
        return (
            <div>
                Click Count: {this.props.count}
                <button type='button' className='navbar-toggle' onClick={this.openApp.bind(this)} />
            </div>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        count: state.count
    };
};

export default connect(mapStateToProps)(App);
