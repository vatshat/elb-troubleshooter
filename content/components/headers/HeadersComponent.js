import React from 'react';
import PropTypes from 'prop-types';
import Toggle from 'react-toggle';

export class TitleComponent extends React.Component {
    
    render() {
        const enable = 'enabled. Please disable capturing to stop the table from auto refreshing in order to view the table';

        return (
            <div>
                <div className='page-header'>
                    <h1 style= {{ color: '#fb9828' }}> Headers </h1>
                </div>
                
                <div className="row">
                    <div className="input-group col-sm-4">
                        <span className="input-group-addon"> URL </span>
                        <input id="msg" type="text" className="form-control" name="msg" placeholder="Please enter url of HTTP endpoint" />
                    </div>
                    <br />
                    { /* <button type="submit" className={'btn btn-default ' + styles.url_submit} onClick={this.reloadHeaders.bind(this)}>Send request</button> */ }
                </div>

                <div id='captureToggle'>
                    { /* 
                        https://reactjs.org/docs/conditional-rendering.html 
                        https://github.com/aaronshaf/react-toggle/blob/master/src/docs/index.js

                    */ }

                    <Toggle
                        id='toggleCaptures'
                        checked={ this.props.toggleCapture }
                        name='toggleCapture'
                        value='yes'
                        onChange={ this.props.handleCaptureToggleChange } />

                    <h5 
                        style={{
                            color: this.props.toggleCapture ? '#f99829' : '#151d27'
                        }}
                        htmlFor='toggleCapture' > 
                        Capturing { this.props.toggleCapture? enable : 'disabled'} 
                    </h5>

                </div>
            </div>
        );
    }
}
TitleComponent.propTypes = {
    handleCaptureToggleChange: PropTypes.func.isRequired,
    toggleCapture: PropTypes.bool.isRequired,
};