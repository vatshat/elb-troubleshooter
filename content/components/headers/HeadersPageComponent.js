import React from 'react';
import BootstrapTable from 'react-bootstrap-table-next';
import paginationFactory from 'react-bootstrap-table2-paginator';
import overlayFactory from 'react-bootstrap-table2-overlay';
import PropTypes from 'prop-types';

export class RefreshButtonComponent extends React.Component {
    
    render() {
        const { handleTableRefresh } = this.props;

        return (
            <button 
                type="button" 
                className="btn btn-default"
                onClick={handleTableRefresh} > 
                Refresh
            </button>

        );
    }
}
RefreshButtonComponent.propTypes = {
    handleTableRefresh: PropTypes.func.isRequired,
};

export class HeadersTableComponent extends React.Component {
    constructor(props){
        super(props);
    }    
    
    render() {
        const columns = [
            {
                dataField: 'id',
                text: 'Request #'
            },
            {
                dataField: 'requestId',
                text: 'HTTP Request ID'
            }, 
            {
                dataField: 'initiator',
                text: 'Initiator'
            },
            {
                dataField: 'timeStamp',
                text: 'Time Stamp'
            }, 
            {
                dataField: 'type',
                text: 'Type'
            }, 
            {
                dataField: 'url',
                text: 'URL'
            }, 
            {
                dataField: 'statusCode',
                text: 'HTTP Response Code'
            }, 
            {
                dataField: 'statusLine',
                text: 'HTTP Response'
            }
        ];

        const NoDataIndication = () => (
            <div className="spinner">
                <div className="rect1" />
                <div className="rect2" />
                <div className="rect3" />
                <div className="rect4" />
                <div className="rect5" />
            </div>
        );

        const CaptionStyle = {
            borderRadius: '0.25em',
            textAlign: 'center',
            color: 'rgb(36, 48, 64)',
            border: '1px solid #fa9728',
            padding: '0.5em'
        };

        const CaptionElement = () => <h1 style={CaptionStyle}>Temp Page</h1>;

        const { data, page, sizePerPage, onTableChange, totalSize, loading } = this.props;
        
        return (
            <div>
                <BootstrapTable
                    remote

                    keyField="id" 
                    data= { data }
                    caption={<CaptionElement />} 
                    columns={ columns } 
                    noDataIndication={ () => <NoDataIndication /> }                     
                    
                    page={ page }
                    loading={ loading }
                    sizePerPage={ sizePerPage }
                    totalSize={ totalSize }
                    onTableChange={ onTableChange }

                    pagination = {
                        paginationFactory({
                            page,
                            sizePerPage,
                            totalSize
                        })
                    }

                    overlay={ 
                        overlayFactory({ 
                            spinner: true, 
                            background: 'rgba(192,192,192,0.3)' 
                        }) 
                    }
                />
            </div>
        );
    }
}

HeadersTableComponent.propTypes = {
    data: PropTypes.array.isRequired,
    page: PropTypes.number.isRequired,
    totalSize: PropTypes.number.isRequired,
    sizePerPage: PropTypes.number.isRequired,
    loading: PropTypes.bool.isRequired,
    onTableChange: PropTypes.func.isRequired,
};