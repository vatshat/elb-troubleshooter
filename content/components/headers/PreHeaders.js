import React from 'react';
import { connect } from 'react-redux';

class Headers extends React.Component {

    componentDidMount() {
        document.addEventListener('click', () => {
            this.props.dispatch({
                type: 'ADD_COUNT'
            });
        });

        /*
        window.history.replaceState({}, document.title, '/');

        fix the unmount of this with 
            https://stackoverflow.com/questions/39094138/reactjs-event-listener-beforeunload-added-but-not-removed
            https://developer.mozilla.org/en-US/Add-ons/WebExtensions/Intercept_HTTP_requests
        */
    }    

    render() {

        if (this.props.preHeadersStoreList) {
            const { preHeadersStoreList } = this.props;
            
            const preHeadersList = preHeadersStoreList.slice(0, 10);

            const PreHeaderComponents = preHeadersList.map((preHeader, index) => {                    

                //const id = preHeader.headers.timeStamp; //Date.now()
                
                const id = index

                return <PreHeader key={id}  {...preHeader } />;
            });
            
            return (
                <div className="row">{PreHeaderComponents}</div>
            );
        }
        else { return ( <div /> ); }
    }
}

const preStyle = {
    wordWrap: 'normal'
};

class PreHeader extends React.Component {
    render() {
        return (
            <pre style={preStyle} className="col-lg-4">
                {JSON.stringify(this.props, null, 2)}
            </pre>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        preHeadersStoreList: state.headers
    };
};

export default connect(mapStateToProps)(Headers);
