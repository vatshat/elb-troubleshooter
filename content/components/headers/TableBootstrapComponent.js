import React from 'react';
import { BootstrapTable, TableHeaderColumn } from 'react-bootstrap-table';
import PropTypes from 'prop-types';

class TableExpandComponent extends React.Component {    
    render() {
        return (
            <pre style={{wordWrap:'normal'}} className='col-sm-12'>
                {JSON.stringify(this.props.data, null, 2)}
            </pre>
        );
    }
}

export class TableBootstrapComponent extends React.Component {
    constructor(props) {
        super(props);
    }

    renderPaginationShowsTotal(start, to, total) {
        return (
            <p style={ { color: 'blue' } }>
            From { start } to { to }, totals is { total }  (its a customize text)
            </p>
        );
    }
    
    expandComponent(row) {
        const propData = typeof row.requestHeaders === 'undefined' ? row.responseHeaders : row.requestHeaders;

        let dataProp = {};

        propData.map((header) => {
            dataProp = {
                ...dataProp,
                [header.name]:header.value
            }            
        });

        return (
            <TableExpandComponent data={ dataProp } />
        );
    }

    render() {
        const options = {
            sizePerPageList: [10, 25, 50, 100],
            paginationShowsTotal: true,
            clearSearch: true
        }

        const selectRow = {
            mode: 'checkbox',
            clickToExpand: true, // click to expand row, default is false
            //clickToSelect: true, // click to select, default is false
            onSelect: this.props.rowSelectHandler,
            onSelectAll: this.props.rowSelectAllHandler
        };

        return (        
            <BootstrapTable data={ this.props.data }
                pagination={ true }
                options={ options }
                bordered={ false }
                search
                searchPlaceholder='Search below table' 

                expandableRow={ () => { return true; } }
                expandComponent={ this.expandComponent }
                expandColumnOptions={ { expandColumnVisible: true, expandColumnBeforeSelectColumn: false } }
                selectRow={ selectRow }>

                <TableHeaderColumn dataField='id' className='td-header-style' isKey={ true } width='5%' headerAlign='center' dataAlign='center'>#</TableHeaderColumn>
                <TableHeaderColumn dataField='requestId' className='td-header-style' headerAlign='center' dataAlign='center'    >Request ID</TableHeaderColumn>
                <TableHeaderColumn dataField='initiator' className='td-header-style' columnTitle={ true } width='15%'>Initiator</TableHeaderColumn>
                <TableHeaderColumn dataField='headerType' className='td-header-style' headerAlign='center' dataAlign='center'>HTTP Type</TableHeaderColumn>
                <TableHeaderColumn dataField='timeStamp' className='td-header-style' columnTitle={ true } width='10%'>Timestamp</TableHeaderColumn>
                <TableHeaderColumn dataField='type' className='td-header-style' headerAlign='center' dataAlign='center'>Type</TableHeaderColumn>
                <TableHeaderColumn dataField='url' className='td-header-style' columnTitle={ true } width='30%'>URL</TableHeaderColumn>
                <TableHeaderColumn dataField='statusCode' className='td-header-style' headerAlign='center' dataAlign='center'>Status Code</TableHeaderColumn>
                <TableHeaderColumn dataField='statusLine' className='td-header-style' columnTitle={ true } width='10%'>Status Line</TableHeaderColumn>
            </BootstrapTable>
        );
    }
}

TableBootstrapComponent.propTypes = {
    data: PropTypes.array.isRequired,
    rowSelectHandler: PropTypes.func.isRequired,
    rowSelectAllHandler: PropTypes.func.isRequired
};