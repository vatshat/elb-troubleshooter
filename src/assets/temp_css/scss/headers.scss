$orange: #cb4b16;
$blue: #2fa4e7;

#nav1:hover,
#nav2:hover,
#nav3:hover,
#nav4:hover {
    background: #151d27;
}

.react-bs-table-sizePerPage-dropdown {
    display: block
}

/*  export button for headers tab */
.export-csv,
.export-csv a {
    background: $blue;
    color: black;
}

.export-csv:hover,
.export-csv a:hover {
    background: black;
    color: $orange;
}
