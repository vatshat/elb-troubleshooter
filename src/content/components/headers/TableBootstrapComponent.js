import React from 'react';
import { BootstrapTable, TableHeaderColumn } from 'react-bootstrap-table';
import { Button, Glyphicon } from 'react-bootstrap';
import { array, func, bool, number} from 'prop-types';
import ReactJson from 'react-json-view'
import { CSVLink } from 'react-csv';

const TableExpandComponent = props => {        
    return (
        <div className='col-sm-4'>
            <ReactJson 
                src={props.data} 
                name="headers"
                displayDataTypes={ false }
            />
        </div>
    );
}

export class ActualBootTableComponent extends React.Component {
    renderPaginationShowsTotal(start, to, total) {
        return (
            <p style={ { color: 'blue' } }>
            From { start } to { to }, totals is { total }  (its a customize text)
            </p>
        );
    }
    
    expandComponent(row) {
        const propData = typeof row.requestHeaders === 'undefined' ? row.responseHeaders : row.requestHeaders;

        let dataProp = {};

        propData.map((header) => {
            dataProp = {
                ...dataProp,
                [header.name]:header.value
            }            
        });

        return (
            <TableExpandComponent data={ dataProp } />
        );
    }

    _overlay() { 
        const { toggleCapture, headersLength } = this.props;

        if (toggleCapture) {
            return (
                <div>                 
                    <h4>For performance reasons, cannot view headers while capturing. Stop capturing to view headers...</h4>
                    <h5 style={{ color: 'rgb(203, 75, 22)' }}>Captured a total of <strong> { headersLength > 0? headersLength : 0 } </strong> HTTP headers so far</h5>
                    <div id="loader" />
                </div>
            );
        }
        else {
            return 'There are no headers captured';
        }        
    }

    render() {
        let options = {
            sizePerPageList: [10, 25, 50, 100],
            paginationShowsTotal: true,
            noDataText: this._overlay(),
            defaultSortName: 'id', // default sort column name
            defaultSortOrder: 'asc' // default sort order
        }

        const selectRow = {
            mode: 'checkbox',
            clickToExpand: true, // click to expand row, default is false
            //clickToSelect: true, // click to select, default is false
            onSelect: this.props.rowSelectHandler,
            onSelectAll: this.props.rowSelectAllHandler
        };

        return (
            <BootstrapTable id='temp' data={ this.props.data }
                pagination={ true }
                options={ options }
                bordered={ false }
                search
                searchPlaceholder='Filter below table based on any column' 

                expandableRow={ () => { return true; } }
                expandComponent={ this.expandComponent }
                expandColumnOptions={ { expandColumnVisible: true, expandColumnBeforeSelectColumn: false } }
                selectRow={ selectRow }
                headerStyle={ { color: this.props.toggleCapture? '#2fa4e7' : 'rgb(203, 75, 22)', whiteSpace: 'normal' } }
            >

                <TableHeaderColumn dataField='id' isKey={ true } width='5%' headerAlign='center' dataAlign='center' dataSort>#</TableHeaderColumn>
                <TableHeaderColumn dataField='requestId' headerAlign='center' dataAlign='center' dataSort>Request ID</TableHeaderColumn>
                <TableHeaderColumn dataField='initiator' columnTitle={ true } width='15%' dataSort>Initiator</TableHeaderColumn>
                <TableHeaderColumn dataField='headerType' headerAlign='center' dataAlign='center' dataSort>HTTP Type</TableHeaderColumn>
                <TableHeaderColumn dataField='timeStamp' columnTitle={ true } width='10%' dataSort>Timestamp</TableHeaderColumn>
                <TableHeaderColumn dataField='type' columnTitle={ true } width='10%' headerAlign='center' dataAlign='center' dataSort>Type</TableHeaderColumn>
                <TableHeaderColumn dataField='url' columnTitle={ true } width='30%' dataSort>URL</TableHeaderColumn>
                <TableHeaderColumn dataField='statusCode' headerAlign='center' dataAlign='center' dataSort>Status Code</TableHeaderColumn>
            </BootstrapTable>
        )
    }
}

const ExportCSVComponent = props => {        
        
    const headers = [
        { label: 'ID', key: 'id' },
        {label: 'Request ID', key: 'requestId'},
        {label: 'Initiator', key: 'initiator'},
        {label: 'Time stamp', key: 'timeStamp'},
        {label: 'Type', key: 'type'},
        {label: 'URL', key: 'url'},
        {label: 'Status Code', key: 'statusCode'},
        {label: 'Status Line', key: 'statusLine'},
        {label: 'Header Type', key: 'headerType'},
        {label: 'HTTP Headers', key: 'headers'},
    ];

    const data = props.data.map((header) => {
        if ( typeof header.responseHeaders === 'object' ) {
            let {responseHeaders, ...headerNew} = header                

            return {
                ...headerNew,
                headers: JSON.stringify(header.responseHeaders)
            }
        }
        else if ( typeof header.requestHeaders === 'object' ) {
            let {requestHeaders, ...headerNew} = header;                

            return {
                ...headerNew,
                headers: JSON.stringify(header.requestHeaders)
            }
        }
    })

    return (             
        <CSVLink 
            headers={headers}
            data={data}
            separator={";"}
            filename={`headers-${ new Date().toISOString() }.csv`}
        >
            <Button className='export-csv'>
                <Glyphicon glyph="export" /> 
                {' '}
                    Export to CSV
            </Button> 
        </CSVLink>
    )
}

const TableBootstrapComponent = props => {
    return (      
        <div>  
            <ExportCSVComponent data={ props.data } />  
            <ActualBootTableComponent { ...props } />                       
        </div>
    );
}

export default TableBootstrapComponent

TableBootstrapComponent.propTypes = {
    data: array.isRequired,
    rowSelectHandler: func.isRequired,
    rowSelectAllHandler: func.isRequired,
    toggleCapture: bool.isRequired, 
    headersLength: number.isRequired,
};