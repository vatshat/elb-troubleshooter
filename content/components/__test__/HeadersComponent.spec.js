import React from 'react'
import renderer from 'react-test-renderer'
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';

import {
    ToggleComponent,
    SelectedHeadersComponent
} from '../headers/HeadersComponent'

import { ActualBootTableComponent } from '../headers/TableBootstrapComponent'

// Snapshot for Home React Component

describe('>>> Snapshots of Components', () => {

    const mockFunction = () => {}

    it('+++ check if toggle button is showing gray', () => {        
        const renderedValue = renderer.create( 
            <ToggleComponent 
                toggleCapture = { false }
                handleCaptureToggleChange = { mockFunction }
            />
        ).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });

    it('+++ check if toggle button is showing blue', () => {
        const renderedValue = renderer.create( 
            <ToggleComponent 
                toggleCapture = { true }
                handleCaptureToggleChange = { mockFunction }
            />
        ).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });

    it('+++ check if header panel is initialized correctly', () => {
        const renderedValue = renderer.create( 
            <SelectedHeadersComponent selectedHeaders = { [] } />
        ).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });

    it('+++ check if header table is initialized correctly', () => {        
        // react tets renderer doesn't work switching to enzyme - https://github.com/AllenFang/react-bootstrap-table/issues/1160


        const wrapper = shallow(
            <ActualBootTableComponent 
                data = { [] }
                headersLength = { 1 }
                rowSelectHandler = { mockFunction }
                rowSelectAllHandler = { mockFunction }
                toggleCapture = { false }
            />
        )
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});