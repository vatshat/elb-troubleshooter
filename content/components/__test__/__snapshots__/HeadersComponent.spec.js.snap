// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> Snapshots of Components +++ check if header panel is initialized correctly 1`] = `
<div
  className="well"
>
  <h4
    style={
      Object {
        "color": "rgb(203, 75, 22)",
        "cursor": "pointer",
      }
    }
  >
    <p>
      No headers selected 
      <small>
        ...select checkbox in any row in the below table to view header details
      </small>
    </p>
  </h4>
</div>
`;

exports[`>>> Snapshots of Components +++ check if header table is initialized correctly 1`] = `
<BootstrapTable
  autoCollapse={
    Object {
      "filter": false,
      "search": false,
      "sort": false,
    }
  }
  bodyContainerClass={null}
  bordered={false}
  cellEdit={
    Object {
      "afterSaveCell": undefined,
      "beforeSaveCell": undefined,
      "blurToEscape": false,
      "blurToSave": false,
      "mode": "none",
      "nonEditableRows": undefined,
    }
  }
  columnFilter={false}
  condensed={false}
  containerClass={null}
  csvFileName="spreadsheet.csv"
  data={Array []}
  deleteRow={false}
  expandColumnOptions={
    Object {
      "expandColumnBeforeSelectColumn": false,
      "expandColumnVisible": true,
    }
  }
  expandComponent={[Function]}
  expandableRow={[Function]}
  exportCSV={false}
  fetchInfo={
    Object {
      "dataTotalSize": 0,
    }
  }
  footer={false}
  headerContainerClass={null}
  headerStyle={
    Object {
      "color": "rgb(203, 75, 22)",
      "whiteSpace": "normal",
    }
  }
  height="100%"
  hover={false}
  id="temp"
  ignoreSinglePage={false}
  insertRow={false}
  keyBoardNav={false}
  multiColumnSearch={false}
  multiColumnSort={1}
  options={
    Object {
      "defaultSortName": "id",
      "defaultSortOrder": "asc",
      "noDataText": "There are no headers captured",
      "paginationShowsTotal": true,
      "sizePerPageList": Array [
        10,
        25,
        50,
        100,
      ],
    }
  }
  pagination={true}
  printable={false}
  renderAlert={true}
  replace={false}
  search={true}
  searchPlaceholder="Filter below table based on any column"
  selectRow={
    Object {
      "clickToExpand": true,
      "mode": "checkbox",
      "onSelect": [Function],
      "onSelectAll": [Function],
    }
  }
  striped={false}
  tableBodyClass={null}
  tableContainerClass={null}
  tableFooterClass={null}
  tableHeaderClass={null}
  trClassName=""
  version="3"
  withoutTabIndex={false}
>
  <TableHeaderColumn
    className=""
    columnClassName=""
    columnTitle={false}
    csvFieldType="string"
    dataAlign="center"
    dataField="id"
    dataSort={true}
    defaultASC={false}
    editColumnClassName=""
    editable={true}
    expandable={true}
    filterFormatted={false}
    headerAlign="center"
    headerTitle={true}
    hidden={false}
    hiddenOnInsert={false}
    invalidEditColumnClassName=""
    isKey={true}
    keyValidator={false}
    searchable={true}
    sortIndicator={true}
    width="5%"
  >
    #
  </TableHeaderColumn>
  <TableHeaderColumn
    className=""
    columnClassName=""
    columnTitle={false}
    csvFieldType="string"
    dataAlign="center"
    dataField="requestId"
    dataSort={true}
    defaultASC={false}
    editColumnClassName=""
    editable={true}
    expandable={true}
    filterFormatted={false}
    headerAlign="center"
    headerTitle={true}
    hidden={false}
    hiddenOnInsert={false}
    invalidEditColumnClassName=""
    isKey={false}
    keyValidator={false}
    searchable={true}
    sortIndicator={true}
    width={null}
  >
    Request ID
  </TableHeaderColumn>
  <TableHeaderColumn
    className=""
    columnClassName=""
    columnTitle={true}
    csvFieldType="string"
    dataAlign="left"
    dataField="initiator"
    dataSort={true}
    defaultASC={false}
    editColumnClassName=""
    editable={true}
    expandable={true}
    filterFormatted={false}
    headerTitle={true}
    hidden={false}
    hiddenOnInsert={false}
    invalidEditColumnClassName=""
    isKey={false}
    keyValidator={false}
    searchable={true}
    sortIndicator={true}
    width="15%"
  >
    Initiator
  </TableHeaderColumn>
  <TableHeaderColumn
    className=""
    columnClassName=""
    columnTitle={false}
    csvFieldType="string"
    dataAlign="center"
    dataField="headerType"
    dataSort={true}
    defaultASC={false}
    editColumnClassName=""
    editable={true}
    expandable={true}
    filterFormatted={false}
    headerAlign="center"
    headerTitle={true}
    hidden={false}
    hiddenOnInsert={false}
    invalidEditColumnClassName=""
    isKey={false}
    keyValidator={false}
    searchable={true}
    sortIndicator={true}
    width={null}
  >
    HTTP Type
  </TableHeaderColumn>
  <TableHeaderColumn
    className=""
    columnClassName=""
    columnTitle={true}
    csvFieldType="string"
    dataAlign="left"
    dataField="timeStamp"
    dataSort={true}
    defaultASC={false}
    editColumnClassName=""
    editable={true}
    expandable={true}
    filterFormatted={false}
    headerTitle={true}
    hidden={false}
    hiddenOnInsert={false}
    invalidEditColumnClassName=""
    isKey={false}
    keyValidator={false}
    searchable={true}
    sortIndicator={true}
    width="10%"
  >
    Timestamp
  </TableHeaderColumn>
  <TableHeaderColumn
    className=""
    columnClassName=""
    columnTitle={true}
    csvFieldType="string"
    dataAlign="center"
    dataField="type"
    dataSort={true}
    defaultASC={false}
    editColumnClassName=""
    editable={true}
    expandable={true}
    filterFormatted={false}
    headerAlign="center"
    headerTitle={true}
    hidden={false}
    hiddenOnInsert={false}
    invalidEditColumnClassName=""
    isKey={false}
    keyValidator={false}
    searchable={true}
    sortIndicator={true}
    width="10%"
  >
    Type
  </TableHeaderColumn>
  <TableHeaderColumn
    className=""
    columnClassName=""
    columnTitle={true}
    csvFieldType="string"
    dataAlign="left"
    dataField="url"
    dataSort={true}
    defaultASC={false}
    editColumnClassName=""
    editable={true}
    expandable={true}
    filterFormatted={false}
    headerTitle={true}
    hidden={false}
    hiddenOnInsert={false}
    invalidEditColumnClassName=""
    isKey={false}
    keyValidator={false}
    searchable={true}
    sortIndicator={true}
    width="30%"
  >
    URL
  </TableHeaderColumn>
  <TableHeaderColumn
    className=""
    columnClassName=""
    columnTitle={false}
    csvFieldType="string"
    dataAlign="center"
    dataField="statusCode"
    dataSort={true}
    defaultASC={false}
    editColumnClassName=""
    editable={true}
    expandable={true}
    filterFormatted={false}
    headerAlign="center"
    headerTitle={true}
    hidden={false}
    hiddenOnInsert={false}
    invalidEditColumnClassName=""
    isKey={false}
    keyValidator={false}
    searchable={true}
    sortIndicator={true}
    width={null}
  >
    Status Code
  </TableHeaderColumn>
</BootstrapTable>
`;

exports[`>>> Snapshots of Components +++ check if toggle button is showing blue 1`] = `
<div
  id="captureToggle"
>
  <div
    className="react-toggle react-toggle--checked"
    onClick={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <div
      className="react-toggle-track"
    >
      <div
        className="react-toggle-track-check"
      >
        <svg
          height="11"
          viewBox="0 0 14 11"
          width="14"
        >
          <title>
            switch-check
          </title>
          <path
            d="M11.264 0L5.26 6.004 2.103 2.847 0 4.95l5.26 5.26 8.108-8.107L11.264 0"
            fill="#fff"
            fillRule="evenodd"
          />
        </svg>
      </div>
      <div
        className="react-toggle-track-x"
      >
        <svg
          height="10"
          viewBox="0 0 10 10"
          width="10"
        >
          <title>
            switch-x
          </title>
          <path
            d="M9.9 2.12L7.78 0 4.95 2.828 2.12 0 0 2.12l2.83 2.83L0 7.776 2.123 9.9 4.95 7.07 7.78 9.9 9.9 7.776 7.072 4.95 9.9 2.12"
            fill="#fff"
            fillRule="evenodd"
          />
        </svg>
      </div>
    </div>
    <div
      className="react-toggle-thumb"
    />
    <input
      checked={true}
      className="react-toggle-screenreader-only"
      id="toggleCaptures"
      name="toggleCapture"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="checkbox"
      value="yes"
    />
  </div>
  <h4
    htmlFor="toggleCapture"
    style={
      Object {
        "color": "#2fa4e7",
      }
    }
  >
    Capturing enabled 
    <small>
      ...Please disable capturing to show the table
    </small>
  </h4>
</div>
`;

exports[`>>> Snapshots of Components +++ check if toggle button is showing gray 1`] = `
<div
  id="captureToggle"
>
  <div
    className="react-toggle"
    onClick={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <div
      className="react-toggle-track"
    >
      <div
        className="react-toggle-track-check"
      >
        <svg
          height="11"
          viewBox="0 0 14 11"
          width="14"
        >
          <title>
            switch-check
          </title>
          <path
            d="M11.264 0L5.26 6.004 2.103 2.847 0 4.95l5.26 5.26 8.108-8.107L11.264 0"
            fill="#fff"
            fillRule="evenodd"
          />
        </svg>
      </div>
      <div
        className="react-toggle-track-x"
      >
        <svg
          height="10"
          viewBox="0 0 10 10"
          width="10"
        >
          <title>
            switch-x
          </title>
          <path
            d="M9.9 2.12L7.78 0 4.95 2.828 2.12 0 0 2.12l2.83 2.83L0 7.776 2.123 9.9 4.95 7.07 7.78 9.9 9.9 7.776 7.072 4.95 9.9 2.12"
            fill="#fff"
            fillRule="evenodd"
          />
        </svg>
      </div>
    </div>
    <div
      className="react-toggle-thumb"
    />
    <input
      checked={false}
      className="react-toggle-screenreader-only"
      id="toggleCaptures"
      name="toggleCapture"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="checkbox"
      value="yes"
    />
  </div>
  <h4
    htmlFor="toggleCapture"
    style={
      Object {
        "color": "#151d27",
      }
    }
  >
    Capturing disabled 
    <small>
      ...Toggle above switch to start capturing
    </small>
  </h4>
</div>
`;
